sentryflow/Dockerfile

```diff
@@ -1,36 +1,35 @@
 # SPDX-License-Identifier: Apache-2.0
+# Copyright 2024 Authors of SentryFlow
 
-### Builder
+FROM golang:1.23 AS builder
 
-FROM golang:1.21-alpine3.17 as builder
+ARG TARGETOS
+ARG TARGETARCH
 
-RUN apk --no-cache update
-RUN apk add --no-cache git clang llvm make gcc protobuf musl-dev
-RUN apk add --update alpine-sdk
+RUN mkdir -p /protobuf/golang
 
-RUN go install github.com/golang/protobuf/protoc-gen-go@latest
-RUN go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@latest
+COPY protobuf/golang /protobuf/golang
 
-RUN mkdir /sentryflow
-RUN mkdir /protobuf
-
-WORKDIR /protobuf
-COPY /protobuf .
+# Required to embed build info into binary.
+COPY .git /.git
 
 WORKDIR /sentryflow
-COPY /sentryflow .
 
-RUN export CGO_ENABLED=1; export CC=gcc;
-RUN go build -o sentryflow
+COPY sentryflow/go.mod .
+COPY sentryflow/go.sum .
+RUN go mod download
+
+COPY sentryflow/cmd cmd/
+COPY sentryflow/pkg pkg/
+COPY sentryflow/main.go main.go
+COPY sentryflow/Makefile Makefile
 
-### Make executable image
+WORKDIR /
 
-FROM alpine:3.17 as sentryflow
+RUN GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} make -C sentryflow build
 
-# RUN echo "@community http://dl-cdn.alpinelinux.org/alpine/edge/community" | tee -a /etc/apk/repositories
-# RUN apk --no-cache update
-# RUN apk add bash
+FROM gcr.io/distroless/static-debian12
```

Comment:
Any particular reason to use distroless instead of static? Especially since it has not CGO Deps